modal/runner.py

```diff
@@ -182,11 +183,13 @@ async def _publish_app(
     classes: dict[str, _Cls],
     name: str = "",  # Only relevant for deployments
     tag: str = "",  # Only relevant for deployments
+    commit_info: Optional[Dict[str, Any]] = None,  # Git commit information
```

Comment:
Perhaps this could by a `TypedDict` or a dataclass since we know the fields we'll have?

Reply:
Or, see below, maybe we should just pass around an `api_pb2.CommitInfo` here.